vm/devices/storage/disk_backend_resources/src/lib.rs

```diff
@@ -33,6 +33,19 @@ impl ResourceId<DiskHandleKind> for DiskWithReservationsHandle {
     const ID: &'static str = "prwrap";
 }
 
+/// Disk handle for a delay disk
+#[derive(MeshPayload)]
+pub struct DelayDiskHandle{
+    /// The underlying disk resource.
+    pub disk: Resource<DiskHandleKind>,
+    /// The delay to add to each I/O operation, in milliseconds.
+    pub delay: u64,
```

Comment:
You might want to use a `mesh::Cell<u64>` so that you can easily update this dynamically.